### DISCUSSION TITLE
Size of pyarmor_runtime.pyd since Pyarmor v8.2.3

### DISCUSSION THREAD

**Discussion by joonis** [dashingsoft/pyarmor#1295] [General]
Since Pyarmor v8.2.3 the size of `pyarmor_runtime.pyd` (windows.x86_64) is 4.7 MB, before it was around 615 KB.

Is this normal?

**Comment by jondy** [comment]
Yes. This is Themida enabled extension, `--enable-themida` is implemented in v8.2.3. It's bigger and slower.

If not enable themida, it should be around 500k.

Look the file size in the `/path/to/pyarmor/cli/core`

**Comment by joonis** [comment]
I did not enable Themida (whether by cfg nor by cli option), but got the huge file...

**Comment by jondy** [comment]
It's a bug, I'll fix it in next release. 

As a quick workaround, apply this patch 
```patch
--- a/pyarmor/cli/core/__init__.py
+++ b/pyarmor/cli/core/__init__.py
@@ -154,7 +154,7 @@ class PyarmorRuntime(object):
         path = __file__.replace('__init__.py', '')
         for x in listdir(path):
             if x.startswith(prefix) and x[-3:] in ('.so', 'pyd'):
-                if extra is None or extra in x.split('.'):
+                if (extra is None and 'themida' not in x.split('.')) or extra in x.split('.'):
                     return x, os_path.join(path, x)
 
         path = os_path.join(path, *plat.split('.'))
```

**Comment by joonis** [comment]
Don't know if it works directly under Windows. But it does not work under Linux with `--platform windows.x86_64`.

**Comment by jondy** [comment]
This patch should not work with cross platform build, anyway, please upgrade to Pyarmor 8.2.4 to fix this issue
